set.mm

```diff
@@ -643671,6 +643722,980 @@ base set if and only if the neighborhoods (convergents) of every point
+       Ridenour, 11-Aug-2023.)  (Proof modification is discouraged.)  Use ~ cp
+       instead.  (New usage is discouraged.) @)
+    rr-cp @p |- E. w A. x e. z ( E. y ph -> E. y e. w ph ) @=
+      ( wex cv wrex wral copab ccoll cvv wcel elex collexd elv wceq nfcv nfcoll
+      wi nfopab1 nfeq2 nfopab2 rexeqf imbi2d ralbid cpcoll ceqsexv2d ) ACFZACEG
+      ZHZTZBDGZIUIACUMABCJZKZHZTZBUMIEUOUOLMDUMLMUMUNLUMLNOPUJUOQZULUQBUMBUJUOB
+      UMUNABCUABUMRSUBURUKUPUIACUJUOCUJRCUMUNABCUCCUMRSUDUEUFABCDUGUH @.
+  @}
+  $)
+
+  ${
+    $d ph x y z $.  $d x y z G $.  $d x y z F $.  $d x y z A $.
+    grucollcld.1 $e |- ( ph -> G e. Univ ) $.
+    grucollcld.2 $e |- ( ph -> F C_ ( G X. G ) ) $.
+    grucollcld.3 $e |- ( ph -> A e. G ) $.
+    $( A Grothendieck universe contains the output of a collection operation
+       whenever its left input is a relation on the universe, and its right
+       input is in the universe.  (Contributed by Rohan Ridenour,
+       11-Aug-2023.) $)
+    grucollcld $p |- ( ph -> ( F Coll A ) e. G ) $=
+      ( vx vy vz ccoll cv wbr cab cscott wcel wa c0 simpr ad2antrr ciun df-coll
+      cgru wral wceq gru0eld eqeltrd wn wex wi neq0 cxp scottss sseldi vex elab
+      breq2 sylib wss ssbrd mpd simprbi gruscottcld expcom exlimiv sylbi impcom
+      brxp syl pm2.61dan ralrimiva gruiun syl3anc syl5eqel ) ABCKHBHLZILZCMZINZ
+      OZUAZDHIBCUBADUCPZBDPZVSDPZHBUDVTDPEGAWCHBAVOBPZQZVSRUEZWCWEWFQZVSRDWEWFS
+      WGBDAWAWDWFETAWBWDWFGTUFUGWFUHZWEWCWHJLZVSPZJUIWEWCUJZJVSUKWJWKJWEWJWCWEW
+      JQZVRWIDAWAWDWJETWLVOWIDDULZMZWIDPZWLVOWICMZWNWLWIVRPWPWLVSVRWIVRUMWEWJSZ
+      UNVQWPIWIJUOVPWIVOCUQUPURWLCWMVOWIACWMUSWDWJFTUTVAWNVODPWOVOWIDDVHVBVIWQV
+      CVDVEVFVGVJVKHBVSDVLVMVN $.
+  $}
+
+$(
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+  Minimal universes
+-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-.-
+$)
+
+  $c MinUni $.
+
+  $( Extend class notation with the class of minimal universes. $)
+  cmnu $a class MinUni $.
+
+  ${
+    $d y z f w v i u $.
+    $( Define the class of minimal universes.  These are the sets which satisfy
+       the predicate on ` y ` in ~ rr-grothprim , except for the ` x e. y `
+       clause.
+
+       A minimal universe is closed under subsets ( ~ mnussd ), powersets
+       ( ~ mnupwd ), and an operation which is similar to a combination of
+       collection and union ( ~ mnuop3d ), from which closure under pairing
+       ( ~ mnuprd ), unions ( ~ mnuunid ), and function ranges ( ~ mnurnd ) can
+       be deduced, from which equivalence with Grothendieck universes
+       ( ~ grumnueq ) can be deduced.  (Contributed by Rohan Ridenour,
+       11-Aug-2023.) $)
+    df-mnu $a
+      |- MinUni = { y | A. z e. y A. f ( ~P z C_ y /\ E. w e. y
+        ( ~P z C_ w /\ A. i e. z
+          ( E. v e. y ( i e. v /\ v e. f )
+            -> E. u e. f ( i e. u /\ U. u C_ w ) )
+        )
+      ) } $.
```

Comment:
```suggestion
            -> E. u e. f ( i e. u /\ U. u C_ w ) ) ) ) } $.
```
We usually use fewer lines. Same below. (most editors do bracket matching, so that shouldn't be a problem)